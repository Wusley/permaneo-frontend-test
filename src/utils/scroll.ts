export default class Scroll {
  static scrollToTop() {
    window.scrollTo( { top: 0, behavior: 'smooth' } )
  }
}
