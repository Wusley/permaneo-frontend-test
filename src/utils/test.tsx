import { configureStore } from '@reduxjs/toolkit'
import { render } from '@testing-library/react'
import { renderHook } from '@testing-library/react'
import '@testing-library/jest-dom'
import Providers from '@/providers'
import { rootReducer, RootState } from '@/redux/store'

const createTestStore = ( preloadedState: Partial<RootState> = {} ) =>
  configureStore( {
    reducer: rootReducer,
    preloadedState,
    middleware: ( getDefaultMiddleware ) => getDefaultMiddleware()
  } )

const renderWithStore = (
  ui: React.ReactElement,
  preloadedState: Partial<RootState> = {},
  options?: unknown
) => {
  const store = createTestStore( preloadedState )

  return {
    ...render( ui, {
      wrapper: ( { children } ) => <Providers store={store}>{children}</Providers>,
      ...options
    } ),
    store
  }
}

const renderHookWithStore = <TProps, TResult>(
  hook: ( props: TProps ) => TResult,
  preloadedState: Partial<RootState> = {}
) => {
  const store = createTestStore( preloadedState )

  return {
    ...renderHook( hook, {
      wrapper: ( { children } ) => <Providers store={store}>{children}</Providers>
    } ),
    store
  }
}

export * from '@testing-library/react'
export { renderWithStore, renderHookWithStore }
