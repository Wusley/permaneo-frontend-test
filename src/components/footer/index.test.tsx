import { renderWithStore } from '@/utils/test'
import Footer from './'

describe( 'Footer Snapshot', () => {
  it( 'matches the snapshot', () => {
    const { container } = renderWithStore( <Footer year={2025} /> )
    expect( container ).toMatchSnapshot()
  } )
} )