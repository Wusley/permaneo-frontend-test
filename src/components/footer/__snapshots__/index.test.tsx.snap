// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Snapshot matches the snapshot 1`] = `
<div>
  <div
    class="css-rdd2rx"
  >
    <span
      class="chakra-text css-i9e4ov"
    >
      CopyrightÂ© 
      2025
      , Grupo Permaneo
    </span>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
